source/cgal/polygon.cc

```diff
@@ -0,0 +1,171 @@
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2025 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/cgal/polygon.h>
+
+#include <deal.II/fe/mapping.h>
+
+#include <deal.II/grid/tria.h>
+
+#ifdef DEAL_II_WITH_CGAL
+
+#  include <CGAL/Boolean_set_operations_2.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+#  ifndef DOXYGEN
+// Template implementations
+namespace CGALWrappers
+{
+  template <typename KernelType>
+  void
+  dealii_cell_to_cgal_polygon(
+    const typename Triangulation<2, 2>::cell_iterator &cell,
+    const Mapping<2, 2>                               &mapping,
+    CGAL::Polygon_2<KernelType>                       &polygon)
+  {
+    const auto &vertices = mapping.get_vertices(cell);
+    polygon.clear();
+    if (cell->reference_cell() == ReferenceCells::Triangle)
+      {
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[0]));
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[1]));
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[2]));
+      }
+    else if (cell->reference_cell() == ReferenceCells::Quadrilateral)
+      {
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[0]));
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[1]));
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[3]));
+        polygon.push_back(
+          CGALWrappers::dealii_point_to_cgal_point<CGAL::Point_2<KernelType>,
+                                                   2>(vertices[2]));
+      }
+    else
+      {
+        DEAL_II_ASSERT_UNREACHABLE();
+      }
+  }
+
+
+
+  template <typename KernelType>
+  void
+  dealii_tria_to_cgal_polygon(const Triangulation<2, 2>   &tria,
+                              CGAL::Polygon_2<KernelType> &polygon)
+  {
+    std::map<unsigned int, unsigned int> face_vertex_indices;
+
+    for (const auto &cell : tria.active_cell_iterators())
```

Comment:
I think this function could use some more commentary. For example, put something before the two loops explaining what each phase of this function does. (The first builds a map from the first to the second vertex index of each boundary edge; the second phase then walks this list from one vertex to the next. But that would have been easier for me to understand with some comments :-) )